File: ch02CaseStudy/Lexi/Lexi/Lexi.Check/Iterator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Lexi.Lexi.Element;

namespace Lexi.Lexi.Check
{
    public abstract class Iterator
    {
    public abstract   Glyph First();
    public abstract Glyph Next();
    public abstract bool isDone();
    public abstract Glyph CurrentItem();
    }
}

File: ch05BehavioralPatterns/Pattern504_Iterator/src/main/java/com/jueee/example01/Iterator.java
package com.jueee.example01;

// 定义抽象迭代器
public interface Iterator<T> {
    
    T first(); // 将游标指向第一个元素

    T next(); // 将游标指向下一个元素

    boolean hasNext(); // 判断是否存在下一个元素

    T currentItem(); // 获取游标指向的当前元素
}
